src/Microsoft.Health.Dicom.Core/Features/Indexing/InstanceReindexer.cs

```diff
@@ -0,0 +1,23 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.Indexing
+{
+    /// <summary>
+    /// Reindexer which reindexes DICOM instance.
+    /// </summary>
+    public class InstanceReindexer : IInstanceReindexer
+    {
+        public Task ReindexInstanceAsync(IReadOnlyList<ExtendedQueryTagStoreEntry> entries, long watermark, CancellationToken cancellationToken = default)
+        {
```

Comment:
IReadOnlyCollection? Can we be consistent on which collection we use?

Reply:
Any reason for `IReadOnlyCollection` > `IReadOnlyList`?
Actually I don't have favor between them, use `IReadOnlyList` just because it provides more capability and we can easily get it from `List<>`

Reply:
They're basically the same. (maybe there are more extensions on `IReadOnlyList`, but didn't compare them all) 

I believe the biggest difference is the addition of an indexer on `IReadOnlyList`. You can also obtain a an `IReadOnlyCollection` from a `List` via https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.asreadonly?view=net-5.0